### ISSUE TITLE
Speed up augmentation of bounding boxes for non-trivial transforms

### ISSUE THREAD

**Issue by de-vri-es** [fizyr/keras-retinanet#68]
Bounding boxes are augmented in a rather expensive way right now for transforms other than flips. This can be improved.

**Comment by wassname** [comment]
I noticed the [imgaug](https://github.com/aleju/imgaug) library that can transform masks and annotation points together. [Example](http://imgaug.readthedocs.io/en/latest/source/examples_bounding_boxes.html).

**Comment by Tracyxzh** [comment]
```
# create image data generator objects
train_image_data_generator = keras.preprocessing.image.ImageDataGenerator(
        horizontal_flip=True,
    )
```
  These code is from train_csv.py . And you means there can have parameter like rotation_range?

**Comment by de-vri-es** [comment]
A small status update:

Progress on the `cheap-augmentation` branch is quite far. All use of `keras.preprocessing.image.ImageDataGenerator` has either been removed (if it was a NOP) or replaced with the new `random_transform_generator(...)`. Applying the transformation is now the responsibility of `keras_retinanet.preprocessing.Generator`.

Unit tests for new code are still missing.

The branch has been rebased on `test-training` since that is probably going to be merged soon (and otherwise there will be conflicts).

**Comment by de-vri-es** [comment]
Closing this in favour of #190.

**Comment by liminghuiv** [comment]
Hi @de-vri-es, @Tracyxzh, @wassname,
Do you have any example on using imgaug to add addtional augmentation to keras-retinanet?

Thanks a lot.

Liming
